Documentation for dynamic scaling of Argo CD application controller. (#67913)
openshift/openshift-docs: modules/gitops-argo-cd-properties.adoc

  * _<AppSync>_ - AppSync is used to control the sync frequency of Argo CD applications
  * _<Sharding.enabled>_ - Enable sharding on the Argo CD Application Controller component. This property is used to manage a large number of clusters to relieve memory pressure on the controller component.
  * _<Sharding.replicas>_ - The number of replicas that will be used to support sharding of the Argo CD Application Controller.

>   * _<Sharding.dynamicScalingEnabled>_ - Enable dynamic scaling of the Argo CD Application Controller component. This property allows the Operator to scale the number of replicas based on the number of clusters being managed by the controller component. Setting this property to `true` overrides the configuration of `Sharding.enabled` and `Sharding.replicas`.

Reviewer: ```suggestion
  * _<Sharding.dynamicScalingEnabled>_ - Enable dynamic scaling of the Argo CD Application Controller component. Use this property to allow the Operator to scale the number of replicas based on the number of clusters the controller component is managing presently. Setting this property to `true` overrides the configuration of the `Sharding.enabled` and `Sharding.replicas` properties.
```

Author: Addressed.

>   * _<Sharding.minShards>_ - The minimum number of repricas of the Argo CD Application Controller.

Reviewer: ```suggestion
  * _<Sharding.minShards>_ - The minimum number of Argo CD Application Controller replicas.
```

Author: Addressed

>   * _<Sharding.minShards>_ - The minimum number of repricas of the Argo CD Application Controller.

Reviewer 2: typo on replicas

Author: Addressed, Thank you

>   * _<Sharding.minShards>_ - The minimum number of repricas of the Argo CD Application Controller.

Author: ACK. Thank you.

Addressed.

>   * _<Sharding.maxShards>_ - The maximum number of repricas of the Argo CD Application Controller.

Author: addressed

>   * _<Sharding.maxShards>_ - The maximum number of repricas of the Argo CD Application Controller.

Reviewer 2: typo on replicas

Author: Addressed.  thank you.

>   * _<Sharding.clusterPerShard>_ - The number of clusters that need to be managed by each shard. When the replica count has reached the `maxShards`, the shards will manage more than one cluster.

Author: ACK.

Addressed.

  * _<Env>_ - Environment to set for the application controller workloads.

|`DisableAdmin`    |Disables the built-in admin user.|`false` |

|`GATrackingID`    |Use a Google Analytics tracking ID.|`__<empty>__` |

> |`GAAnonymizeusers`    |Enable hashed usernames sent to google analytics.|`false` |

Reviewer 3: Google and analytics should have initial cap, like in the previous line.

Author: Noted.
Addressed. thank you.
